ckanext/switzerland/helpers/format_utils.py

```diff
@@ -32,59 +32,51 @@ def ogdch_get_format_mapping():
-    if not resource_format and resource.get('format'):
-        resource_format = resource['format'].split('/')[-1].lower()
-
-    # check if 'media_type' or 'format' can be mapped
-    has_format = (_map_to_valid_format(
-        resource_format,
-        format_mapping
-    ) is not None)
-
-    # if the fields can't be mapped,
-    # try to parse the download_url as a last resort
-    if not has_format and resource.get('download_url'):
-        path = urlparse.urlparse(resource['download_url']).path
-        ext = os.path.splitext(path)[1]
-        if ext:
-            resource_format = ext.replace('.', '').lower()
-
-    mapped_format = _map_to_valid_format(
-        resource_format,
-        format_mapping
-    )
-    if mapped_format:
-        # if format could be successfully mapped write it to format field
-        resource['format'] = mapped_format
-    elif not resource.get('download_url'):
-        resource['format'] = 'SERVICE'
-    else:
-        # else return empty string (this will be indexed as N/A)
-        resource['format'] = ''
-
-    # if format could not be mapped and media_type exists use this value  # noqa
-    if (not resource.get('format') and resource.get('media_type')):
-        resource['format'] = resource['media_type'].split('/')[-1]
+format_mapping = ogdch_get_format_mapping()
+
+
+def prepare_resource_format(resource):
+    """the format of the resource is derived in the following way:
+    the first matching case is taken
+    - case 1 media_type is set: it is derived from the media type
+    - case 2 format is set: check whether the format is already mapped
+             or whether it can be mapped
+    - case 3 download url is set: check if the format can be derived
+             from the file extention, otherwise set it as blank
+    - case 4 no download url: set the format to SERVICE
+    """
+    media_type = resource.get('media_type')
+
+    if media_type:
+        resource_format = _get_format_from_media_type(media_type)
+        if resource_format:
+            resource['format'] = resource_format
+            return resource
+
+    format = resource.get('format')
+    resource_format = _map_to_valid_format(format)
+    if resource_format:
+        resource['format'] = resource_format
+        return resource
+
+    if resource.get('download_url'):
+        resource_format = _get_format_from_path(resource)
+        if resource_format:
+            resource['format'] = resource_format
+        else:
+            resource['format'] = 'SERVICE'
+        return resource
 
+    resource['format'] = ''
     return resource
 
 
-def prepare_formats_for_index(resources, format_mapping):
+def prepare_formats_for_index(resources):
     """generates a set with formats of all resources"""
     formats = set()
     for r in resources:
         resource = prepare_resource_format(
-            resource=r,
-            format_mapping=format_mapping
+            resource=r
```

Comment:
Due to the different deriving of the format mapping, the arguments for the function call changed.